src/AzureAppConfigurationImpl.ts

```diff
@@ -291,17 +417,73 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
                 this.#refreshTimer.backoff();
                 throw error;
             }
+            return Promise.resolve(true);
+        }
 
-            // successfully refreshed, run callbacks in async
-            for (const listener of this.#onRefreshListeners) {
-                listener();
+        return Promise.resolve(false);
+    }
+
+    /**
+     * Refresh feature flags.
+     * @returns true if feature flags are refreshed, false otherwise.
+     */
+    async #refreshFeatureFlags(): Promise<boolean> {
+        // if still within refresh interval/backoff, return
+        if (!this.#featureFlagRefreshTimer.canRefresh()) {
+            return Promise.resolve(false);
+        }
+
+        // check if any feature flag is changed
+        let needRefresh = false;
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter,
+                pageEtags: selector.pageEtags
+            };
+            const pageIterator = listConfigurationSettingsWithTrace(
+                this.#requestTraceOptions,
+                this.#client,
+                listOptions
+            ).byPage();
+
+            let pageCount = 0;
+            for await (const page of pageIterator) {
+                if (page._response.status === 200) { // created or changed
+                    needRefresh = true;
+                    break;
+                }
+                // unchanged, check next page
+                pageCount++;
+            }
+
+            if (pageCount !== selector.pageEtags?.length) {
+                needRefresh = true; // page count changed indicating feature flags are added or deleted
+            }
+
+            if (needRefresh) {
+                break; // short-circuit if result from any of the selectors is changed
             }
         }
+
+        if (needRefresh) {
+            try {
+                await this.#loadFeatureFlags();
+                this.#featureFlagRefreshTimer.reset();

```

Comment:
We should reset the refresh timer even if nothing changed on the server. This is to maintain `refresh interval` gap between refresh attempts. Same for regular kv.

Reply:
b1cf809